src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,149 @@
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.OrderDao;
+import mate.academy.internetshop.exceptions.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Order;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.util.ConnectionUtil;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public Order create(Order element) {
+        String query = "INSERT INTO orders (user_id) VALUES (?)";
+        String query2 = "INSERT INTO orders_products (product_id, order_id) VALUES (?, ?)";
+        Long key;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, element.getUserId());
+            statement.execute();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            generatedKeys.next();
+            key = generatedKeys.getLong(1);
+            for (Product product : element.getProducts()) {
+                statement = connection.prepareStatement(query2);
+                statement.setLong(1, product.getId());
+                statement.setLong(2, key);
+                statement.execute();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create order in DataBase");
+        }
+        return get(key).get();
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders "
+                + "LEFT JOIN orders_products ON orders.order_id = orders_products.order_id "
+                + "LEFT JOIN products ON orders_products.product_id = products.product_id "
+                + "WHERE orders.order_id = ?";
+        Order order;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.next();
+            order = getOrderFromResultSet(resultSet);
+            order = getProductsFromOrder(order);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order from DataBase");
+        }
+        return Optional.of(order);
+    }
+
+    @Override
+    public List<Order> getAll() {
+        String query = "SELECT * FROM orders";
+        List<Order> orders = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Order order = getOrderFromResultSet(resultSet);
+                orders.add(getProductsFromOrder(order));
```

Comment:
can we use get() method here?

Reply:
We can because we need an object (order) with products, no??

Reply:
Or you mean to write something like:
Long userId = resultSet.getLong("user_id");
Long orderId = resultSet.getLong("order_id");
Why do we need it if we will write one redundant variable or it will be faster?

Reply:
Do you mean to write something like that?
Long userId = resultSet.getLong("user_id");
Long orderId = resultSet.getLong("order_id");